src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2196,7 +2222,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             throw SemanticAbort();
         }
         ASR::ttype_t *type = ASRUtils::TYPE(
-            ASR::make_Logical_t(al, x.base.base.loc, 4, nullptr, 0));
+            ASR::make_Logical_t(al, x.base.base.loc, 1, nullptr, 0));
```

Comment:
Currently we use logical with kind=4 everywhere, so let's stick to that. If we are going to change it from 4 to 1, then let's do it as a separate PR and do it everywhere. I currently don't know the pros/cons.

Reply:
Shifting from 4 to 1 will help in using lesser memory in the backend. That's one advantage. However, I agree that we should do this separately otherwise the diff gets confusing here.

Reply:
See https://github.com/lcompilers/lpython/pull/476.

Reply:
Note that how the backend actually implements `logical` with `kind=4` is a separate thing, and it can still implement it using just one byte --- in fact that's exactly what we do: https://github.com/lcompilers/lpython/blob/e5af88ae1454236105c416e08dcd9a0ae69b6f27/src/libasr/codegen/asr_to_llvm.cpp#L3375.